Update requirements checking and add a test

LKS90 opened the pull request:
Tests are still passing, but nothing ported yet.

ultimate_cron_job_load_all() and _ultimate_cron_get_class('job')

need a new version / equivalent.

LKS90 committed:
update requirements checking, add test for it

LKS90 commented:
Passes, travis failed with one build.

Berdir commented on src/Plugin/ultimate_cron/Scheduler/Crontab.php:
```diff
@@ -159,14 +159,15 @@ public function isBehind(CronJob $job) {
     if (!$job_last_ran) {
-      $registered = variable_get('ultimate_cron_hooks_registered', array());
-      if (empty($registered[$job->id()])) {
+      $registered = \Drupal::configFactory()->getEditable('ultimate_cron_hooks_registered');
```
I have no idea what this was before, but there is no such configuration in 8.x-1.x.

Berdir commented on ultimate_cron.install:
```diff
@@ -209,7 +207,7 @@ function ultimate_cron_requirements($phase) {
         $requirements['description'][] = t(
           'Some jobs are behind their schedule. Please check if !link is running properly.', array(
-          '!link' => l(t('cron'), 'admin/config/system/cron')
+          '!link' =>  'admin/config/system/cron',
```
All links need to use Url::fromRoute().

Berdir replied:
Or Drupal::l(). But it's quite common for transltable strings to include the <a> tags and just inject the url.

Berdir commented:
Also, missing tests. There's an issue for past that adds a requirements test. This can do something similar.

LKS90 commented:
Regarding the requirements test:

https://www.drupal.org/node/2511822

LKS90 commented on ultimate_cron.install:
```diff
@@ -209,7 +208,7 @@ function ultimate_cron_requirements($phase) {
         $requirements['description'][] = t(
           'Some jobs are behind their schedule. Please check if !link is running properly.', array(
-          '!link' => l(t('cron'), 'admin/config/system/cron')
+          '!link' => \Drupal::l('Cron', Url::fromRoute('system.cron')),
```
readd t('Cron')?

Berdir replied:
Yes.

LKS90 commented:
Alright, I moved the test to a WebTestBase test, makes things easier. The route rebuilidng did help the last time, but then I just had more problems because the cron_key was an empty string (I tried resetting the state cache, didn't help). After moving the test to CronJobInstall (which makes sense, considering the requirements are tested in ultimate_cron.install) the route rebuilding is not necessary and I have a cron key.

I also had to fix the isBehind function, which was looking for $job->disabled which doesn't exist anymore, changed it to $job->status().

Berdir commented:
Nice, merged.
